langlib/lang.value/src/main/java/org/ballerinalang/langlib/value/FromJsonWithType.java

```diff
@@ -97,9 +97,24 @@ private static Object convert(Object value, Type targetType, List<TypeValuePair>
 
         unresolvedValues.add(typeValuePair);
 
-        List<Type> convertibleTypes = TypeConverter.getConvertibleTypesFromJson(value, targetType, new ArrayList<>());
+        List<String> errors = new ArrayList<>();
+        List<Type> convertibleTypes = TypeConverter.getConvertibleTypesFromJson(value, targetType,
+                null, new ArrayList<>(), errors);
         if (convertibleTypes.isEmpty()) {
-            throw createConversionError(value, targetType);
+            if (errors.isEmpty()) {
```

Comment:
Shall we move this part to `CloneUtils` and reuse, since this is duplicated in another place?